src/CommunityToolkit.Maui/ImageSources/GravatarImageSource.shared.cs

```diff
@@ -0,0 +1,239 @@
+	public static readonly BindableProperty ImageProperty = BindableProperty.Create(nameof(Image), typeof(DefaultImage), typeof(GravatarImageSource), defaultValue: DefaultImage.MysteryPerson, propertyChanged: OnDefaultImagePropertyChanged);
+
+	/// <summary>The backing store for the <see cref="ParentHeight" /> bindable property.</summary>
+	internal static readonly BindableProperty ParentHeightProperty = BindableProperty.Create(nameof(ParentHeight), typeof(int), typeof(GravatarImageSource), defaultValue: defaultSize, propertyChanged: OnSizePropertyChanged);
+
+	/// <summary>The backing store for the <see cref="ParentWidth" /> bindable property.</summary>
+	internal static readonly BindableProperty ParentWidthProperty = BindableProperty.Create(nameof(ParentWidth), typeof(int), typeof(GravatarImageSource), defaultValue: defaultSize, propertyChanged: OnSizePropertyChanged);
+
+	const int cancellationTokenSourceTimeout = 737;
+	const string defaultGravatarImageAddress = "https://www.gravatar.com/avatar/";
+	const int defaultSize = 80;
+	static readonly HttpClient singletonHttpClient = new();
+	int gravatarSize;
+	CancellationTokenSource? tokenSource;
+
+	/// <summary>Initializes a new instance of the <see cref="GravatarImageSource"/> class.</summary>
+	public GravatarImageSource()
+	{
+		Stream = new Func<CancellationToken, Task<Stream>>(cancelationToken => DownloadStreamAsync(Uri, cancelationToken));
+		Uri = new Uri(defaultGravatarImageAddress);
+	}
+
+	/// <summary>Gets or sets a <see cref="TimeSpan"/> structure that indicates the length of time after which the image cache becomes invalid.</summary>
+	public TimeSpan CacheValidity
+	{
+		get => (TimeSpan)GetValue(CacheValidityProperty);
+		set => SetValue(CacheValidityProperty, value);
+	}
+
+	/// <summary>Gets or sets a Boolean value that indicates whether caching is enabled on this <see cref="GravatarImageSource"/> object.</summary>
+	public bool CachingEnabled
+	{
+		get => (bool)GetValue(CachingEnabledProperty);
+		set => SetValue(CachingEnabledProperty, value);
+	}
+
+	/// <summary>Gets or sets the email address.</summary>
+	public string? Email
+	{
+		get => (string)GetValue(EmailProperty);
+		set => SetValue(EmailProperty, value);
+	}
+
+	/// <summary>Gets or sets the default image.</summary>
+	public DefaultImage Image
+	{
+		get => (DefaultImage)GetValue(ImageProperty);
+		set => SetValue(ImageProperty, value);
+	}
+
+	/// <summary>Gets a value indicating whether the control email is empty.</summary>
+	public override bool IsEmpty => string.IsNullOrEmpty(Email);
+
+	/// <summary>Gets or sets the URI for the image to get.</summary>
+	[System.ComponentModel.TypeConverter(typeof(UriTypeConverter))]
+	public Uri Uri { get; set; }
+
+	/// <summary>Gets or sets the parent height.</summary>
+	internal int ParentHeight => (int)GetValue(ParentHeightProperty);
+
+	/// <summary>Gets or sets the parent width.</summary>
+	internal int ParentWidth => (int)GetValue(ParentWidthProperty);
```

Comment:
where is it used?

Reply:
These are bound to the parents `Width` and `Height` bindable properties, which then allows us to assign the `propertyChanged` methods.

Reply:
Could you please update samples to show where they are used?

Am I right, that these properties are used to set the Url parameter of the Gravatar URL?

Reply:
@VladislavAntonyuk What kind of samples are you looking for?  Not that the `Button` sample does not have any `WidthRequest` or `HeightRequest` properties, but the core .NET MAUI code updates the width one time and the height two times, resulting in three `OnSizePropertyChanged` calls (with a built-in delay of 1000ms between the two height changes).

Reply:
ImageSource shouldn't change the size of the parent. It just returns the image. Then the parent control (image, button etc) adjusts the image (stretch, zoom etc). If I set the parent the size and image source returns the image of another size, then image source shouldn't change the parent size.
Don't think I'm rude, ss I already told. we should better cache the response for a few seconds and return it if the parameters are the same, rather than trying to add magic delays and properties to reduce the number of requests.

Reply:
Always happy to have a discussion and exchange of ideas and opinions, this is what makes it a great "community" effort :-D

We don't change the size of the parent just GET the parent's size to use as the size request for the ImageSource.  This is instead of having an additional `Size` property, which I think makes it more developer friendly.

Reply:
ok, probably the summary made me a bit confused, could you please remove "sets"?

I would still suggest having a Size property. just to provide users with more flexibility. On the other hand, I would keep only 1 approach.

Reply:
Entirely my fault that the summary was confusing. Originally, I implemented as a getter and setter and documented.  However, it became clear it should ONLY be a getter, and I failed to update the summary.
Summary updated.

Reply:
The original XF implementation provided a `Size` property.  However, I personally felt that as an `ImageSource` it should naturally take it's sizing from the parent.  Normally a `ImageSource` would just get the image, whatever size the source image.  The parent would then be responsible for the stretch, zoom, aspect, etc.

For a standard `ImageSource` if the image was 8000x8000, this is quite a stream to download, to then have the parent scale potentially to 80x80! 

Gravatar however, returns sized images (default is 80x80) and without us requesting an image size matching the parent, it can result in a blurry low-res image. For example, the parent being 500x500 and we request the size of 80x80, it would be terribly low-res.

I would be more than happy to add a `Size` property, but I do feel that getting from the parent is the more elegant and developer friendly implementation, but I may be favouring my implementation.

Reply:
Please vote:
❤️ - Get size automatically from the parent (current solution)
👍- Use Size property (allows user explicitly set the size of the image)

Reply:
Circling back to this, I think we _have to_ rely on the parent's size.

The `StreamImageSource` inherits from `Microsoft.Maui.Controls.Element` which doesn't have the concept of `Size`, `Height`, `Width`, or `X` and `Y`, etc.

In other words, `GravatarImageSource` is not a visual element (e.g. it's not a `View`), so it must rely on the size of the `View` that implements it.

Reply:
I think taking the size of the parent makes sense for now as it will likely cover the majority of scenarios. If someone wants to explicitly define a size then I think it's fine to wait for them to ask.